spec/features/user_order_show_page_spec.rb

```diff
@@ -52,5 +52,39 @@
       expect(page).to have_content("Total items in order: 6")
       expect(page).to have_content("Grand total: $14.00")
     end
+
+    it 'can cancel the order' do
+      merch = create(:merchant)
+      user_1 = create(:user)
+      order_1 = create(:order, user: user_1)
+      item_1 = create(:item, inventory: 10, user: merch)
+      item_2 = create(:item, inventory: 10, user: merch)
+      item_3 = create(:item, inventory: 10, user: merch)
+
+
+
+      order_item_1 = create(:fulfilled_order_item, order: order_1, item: item_1,  quantity: 1, order_price: 100)
+      order_item_2 = create(:fulfilled_order_item, order: order_1, item: item_2, quantity: 2, order_price: 200)
+      order_item_3 = create(:fulfilled_order_item, order: order_1, item: item_3,  quantity: 3, order_price: 300)
+
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user_1)
+
+      visit profile_order_path(order_1)
+
+      expect(page).to have_link("Cancel Order")
+
+      click_on "Cancel Order"
+
+      expect(current_path).to eq(profile_path(user_1))
+      expect(page).to have_content("Your order has been cancelled")
+
```

Comment:
We can only test for the the status to be cancelled with in a with in block because the test will default to the originally created items. I can add it to the within block but it seems though it might be a bit redundant.